flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/DynamicPartitionPruningUtils.java

```diff
@@ -45,153 +52,177 @@
-                || supportDynamicPartitionPruning(join, false);
-    }
-
-    /**
-     * For the input join node, judge whether the join left side and join right side meet the
-     * requirements of dynamic partition pruning. Fact side in left or right is clear. If meets the
-     * requirements, return true.
-     */
-    public static boolean supportDynamicPartitionPruning(Join join, boolean factInLeft) {
-        if (!ShortcutUtils.unwrapContext(join)
-                .getTableConfig()
-                .get(OptimizerConfigOptions.TABLE_OPTIMIZER_DYNAMIC_FILTERING_ENABLED)) {
-            return false;
-        }
-        // Now dynamic partition pruning supports left/right join, inner and semi join. but now semi
-        // join can not join reorder.
-        if (join.getJoinType() == JoinRelType.LEFT) {
-            if (factInLeft) {
-                return false;
-            }
-        } else if (join.getJoinType() == JoinRelType.RIGHT) {
-            if (!factInLeft) {
-                return false;
-            }
-        } else if (join.getJoinType() != JoinRelType.INNER
-                && join.getJoinType() != JoinRelType.SEMI) {
-            return false;
-        }
-
-        JoinInfo joinInfo = join.analyzeCondition();
-        if (joinInfo.leftKeys.isEmpty()) {
-            return false;
-        }
-        RelNode left = join.getLeft();
-        RelNode right = join.getRight();
-
-        // TODO Now fact side and dim side don't support many complex patterns, like join inside
-        // fact/dim side, agg inside fact/dim side etc. which will support next.
-        return factInLeft
-                ? isDynamicPartitionPruningPattern(left, right, joinInfo.leftKeys)
-                : isDynamicPartitionPruningPattern(right, left, joinInfo.rightKeys);
-    }
-
-    private static boolean isDynamicPartitionPruningPattern(
-            RelNode factSide, RelNode dimSide, ImmutableIntList factSideJoinKey) {
-        return isDimSide(dimSide) && isFactSide(factSide, factSideJoinKey);
-    }
-
-    /** make a dpp fact side factor to recurrence in fact side. */
-    private static boolean isFactSide(RelNode rel, ImmutableIntList joinKeys) {
-        DppFactSideFactors factSideFactors = new DppFactSideFactors();
-        visitFactSide(rel, factSideFactors, joinKeys);
-        return factSideFactors.isFactSide();
-    }
-
     /**
      * Judge whether input RelNode meets the conditions of dimSide. If joinKeys is null means we
      * need not consider the join keys in dim side, which already deal by dynamic partition pruning
      * rule. If joinKeys not null means we need to judge whether joinKeys changed in dim side, if
      * changed, this RelNode is not dim side.
      */
-    private static boolean isDimSide(RelNode rel) {
+    public static boolean isDimSide(RelNode rel) {
         DppDimSideFactors dimSideFactors = new DppDimSideFactors();
         visitDimSide(rel, dimSideFactors);
         return dimSideFactors.isDimSide();
```

Comment:
Rename `DppDimSideFactors` to `DppDimSideChecker` and extract the logic into `DppDimSideChecker`  ?

Reply:
> Rename `DppDimSideFactors` to `DppDimSideChecker` and extract the logic into `DppDimSideChecker` ?

Done!